examples/speech-to-text/agent.py

```diff
@@ -16,23 +16,33 @@ async def entrypoint(job: JobContext):
     stt = STT()
     stt_stream = stt.stream()
 
-    async def process_track(track: rtc.Track):
-        audio_stream = rtc.AudioStream(track)
-        async for audio_frame_event in audio_stream:
-            stt_stream.push_frame(audio_frame_event.frame)
+    audio_stream_future = asyncio.Future[rtc.AudioStream]()
 
-    def on_track_subscribed(
-        track: rtc.Track,
-        pub: rtc.TrackPublication,
-        rp: rtc.RemoteParticipant,
-    ):
-        asyncio.create_task(process_track(track))
+    def on_track_subscribed(track: rtc.Track, *_):
+        if track.kind == rtc.TrackKind.KIND_AUDIO:
+            audio_stream_future.set_result(rtc.AudioStream(track))
```

Comment:
i think it is a bad idea to use a future here, it'll throw if it is already done

Reply:
Hrmm yeah I guess for stt we can handle multiple tracks but for kitt we really can only handle one track. So I can wrap it in try/except and log an error if there are other tracks that come. Wdyt?

Reply:
if you can find something simpler it would be cool, the goal here is to have the minimum amount of code
why not starting the function here?

Reply:
the entry point doesn't have to be long-lived

Reply:
ahh really? I was under the impression that when it exits the agent is finished. But hrmm the tasks ought to go somewhere right. Fire-and-forget is probably not the best example code.

Reply:
The important thing is to keep a reference to it. These tasks will be cancelled when the job process ends

Reply:
wdyt about what I just updated it to? not simple enough?